src/matrix/verification/SAS/channel/MockChannel.ts

```diff
@@ -0,0 +1,139 @@
+import type {ILogItem} from "../../../../lib";
+import {createCalculateMAC} from "../mac";
+import {VerificationCancelledError} from "../VerificationCancelledError";
+import {IChannel} from "./Channel";
+import {CancelTypes, VerificationEventTypes} from "./types";
+import anotherjson from "another-json";
+
+interface ITestChannel extends IChannel {
+    setOlmSas(olmSas): void;
+}
+
+export class MockChannel implements ITestChannel {
+    public sentMessages: Map<string, any> = new Map();
+    public receivedMessages: Map<string, any> = new Map();
+    public initiatedByUs: boolean;
+    public startMessage: any;
+    public isCancelled: boolean = false;
+    private olmSas: any;
+    public ourUserDeviceId: string;
+
+    constructor(
+        public otherUserDeviceId: string,
+        public otherUserId: string,
+        public ourUserId: string,
+        private fixtures: Map<string, any>,
+        private deviceTracker: any,
+        public id: string,
+        private olm: any,
+        startingMessage?: any,
+    ) {
+        if (startingMessage) {
+            const eventType = startingMessage.content.method ? VerificationEventTypes.Start : VerificationEventTypes.Request;
+            this.id = startingMessage.content.transaction_id;
+            this.receivedMessages.set(eventType, startingMessage);
+        }
+    }
+
+    async send(eventType: string, content: any, _: ILogItem) {
+        if (this.isCancelled) {
+            throw new VerificationCancelledError();
+        }
+        Object.assign(content, { transaction_id: this.id });
+        this.sentMessages.set(eventType, {content});
+    }
+
+    async waitForEvent(eventType: string): Promise<any> {
+        if (this.isCancelled) {
+            throw new VerificationCancelledError();
+        }
+        const event = this.fixtures.get(eventType);
+        if (event) {
+            this.receivedMessages.set(eventType, event);
+        }
+        else {
+            await new Promise(() => {});
+        }
+        if (eventType === VerificationEventTypes.Mac) {
+            await this.recalculateMAC();
+        }
+        if(eventType === VerificationEventTypes.Accept && this.startMessage) {
+        }
+        return event;
+    }
+
+    private recalculateCommitment() {
+        const acceptMessage = this.acceptMessage?.content;
+        if (!acceptMessage) {
+            return;
+        }
+        const {content} = this.startMessage;
+        const {content: keyMessage} = this.fixtures.get(VerificationEventTypes.Key);
+        const key = keyMessage.key;
+        const commitmentStr = key + anotherjson.stringify(content);
+        const olmUtil = new this.olm.Utility();
+        const commitment = olmUtil.sha256(commitmentStr);
+        olmUtil.free();
+        acceptMessage.commitment = commitment;
```

Comment:
I'm a bit confused why we modify the sent or received event here and set the commitment on it?

Reply:
This is for testing where we use hard coded messages. The commitment on them needs to change based on your public key which is generated at random by olm.

Reply:
Oh, I missed that this is in the `MockChannel`, not in the real one!